### PR TITLE
Feature/selc 1193

### PR DESCRIPTION
**Pull request by andrea-putzu**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala [hunk 1]
```diff
@@ -53,10 +53,18 @@ class ProcessApiServiceImpl(
...
-    mailer.sendMail(mailTemplate)(addresses, file, onboardingMailParameters)("onboarding-contract-email")
+  private def sendOnboardingMail(
+    addresses: Seq[String],
+    file: File,
+    productName: String,
+    onboardingMailParameters: Map[String, String]
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
+    mailer.sendMail(mailTemplate.copy(subject = productName + ": Accordo di Adesione"))(
+      addresses,
+      productName + "_accordo_adesione.pdf",
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala]:
```suggestion
      s"${productName}_accordo_adesione.pdf",
```

File: src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala [hunk 2]
```diff
@@ -53,10 +53,18 @@ class ProcessApiServiceImpl(
...
-    implicit contexts: Seq[(String, String)]
-  ): Future[Unit] = {
-    mailer.sendMail(mailTemplate)(addresses, file, onboardingMailParameters)("onboarding-contract-email")
+  private def sendOnboardingMail(
+    addresses: Seq[String],
+    file: File,
+    productName: String,
+    onboardingMailParameters: Map[String, String]
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] = {
+    mailer.sendMail(mailTemplate.copy(subject = productName + ": Accordo di Adesione"))(
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala]:
```suggestion
    mailer.sendMail(mailTemplate.copy(subject = s"${productName}: Accordo di Adesione"))(
```

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala]: These are strings, so there's no harm, but using string interpolation is actually a good practice. See: https://nrinaudo.github.io/scala-best-practices/tricky_behaviours/string_concatenation.html

**Review by galales — APPROVED** [review]

**Comment by TonioGela** [comment]
The CI is broken since the tests are flaky. Do you want me to upload the report?

**Comment by TonioGela** [comment]
CI is now green on first try

**Review by TonioGela — APPROVED** [review]

**Pull request merged**